src/util/codec/datum.rs

```diff
@@ -1341,4 +1395,55 @@ mod test {
             assert_eq!(res_y, exp_y);
         }
     }
+
+    #[test]
+    fn test_cast_as_json() {
+        let tests = vec![
+            (Datum::I64(1), "1.0"),
+            (Datum::F64(3.3),"3.3"),
+            (Datum::Bytes(br#""Hello,world""#.to_vec()), r#""Hello,world""#),
+            (Datum::Bytes(b"[1, 2, 3]".to_vec()), "[1, 2, 3]"),
+            (Datum::Bytes(b"{}".to_vec()), "{}"),
+            (Datum::I64(1), "true"),
```

Comment:
It's the same with `tidb` https://github.com/pingcap/tidb/blob/master/util/types/datum.go#L1227 @hicqu